MarkdownKit/Sources/Common/Elements/List/MarkdownList.swift

```diff
@@ -42,14 +42,16 @@ open class MarkdownList: MarkdownLevelElement {
 
     let calcFont = font ?? MarkdownParser.defaultFont
     let nonListSectionRange = NSRange(location: 0, length: range.length)
```

Comment:
@kublaios do you recall what the motivation for this `nonListSectionRange` was?  The range used is outside the line of text being actively formatted (`location: 0`) which caused some slight formatting glitches when I was testing it out

Reply:
this seems to be why the unit tests are crashing:
> /Users/runner/work/MarkdownKit/MarkdownKit/MarkdownKit/Sources/Common/Elements/List/MarkdownList.swift:47: error: -[MarkdownKitTests.Tests testParseListWithLotsOfSpacesAfterIndicator] : NSMutableRLEArray objectAtIndex:effectiveRange:: Out of bounds (NSRangeException)